boggle.js

```diff
@@ -11,21 +11,196 @@ function newBoggleBoard() {
  * Prints out a boggle board.
  */
 function printBoggleBoard(boggleBoard) {
+  /*
+  let includesQu = false;
+  for (let i = 0; i < boggleBoard.length; i++) {
+    for (let j = 0; j < boggleBoard[i].length; j++) {
+      if (boggleBoard[i][j] === 'Q') {
+        boggleBoard[i][j] = 'Qu';
+        includesQu = true;
+      }
+    }
+  }
+  */
+  /*
+  if (includesQu) {
+    for (let i = 0; i < boggleBoard.length; i++) {
+      for (let j = 0; j < boggleBoard[i].length; j++) {
+        boggleBoard[i][j] = boggleBoard[i][j].padEnd(2);
+      }
+    }
+  }
+  */
+  function formatRow(row) {
+    let formattedRow = [];
+    for (let letter of row) {
+      if (letter === 'Q') {
+        formattedRow.push('Qu');
+      } else {
+        formattedRow.push(letter.padEnd(2));
+      }
+    }
+    return formattedRow;
+  }
+
+  console.log();
   for (let row of boggleBoard) {
-    console.log(row.join(' '));
+    console.log(formatRow(row).join(' '));
   }
+  console.log();
+}
+
+// Function to shuffle array, courtesy of Lily
+function shuffle(array) {
```

Comment:
While this function appears to shuffle, not all shufflings are equally likely. Try shuffling the same 3-element array a few thousand times and using the code from your text analysis program to generate a table of frequencies.

You'll see that certain arrangements are much more common.